From 543560f52347ddca972c62f3e8eeddb1cee813cf Mon Sep 17 00:00:00 2001
From: Santhosh Kumar <santhosh.kumar.selvaraj@qt.io>
Date: Mon, 22 Sep 2025 12:27:51 +0200
Subject: [PATCH 069/239] Mark security header for the qt labs sharedimage

The image loader (QuickSharedImageLoader) uses shared memory
(QSharedMemory) but it shouldn't trigger security issue from the
framework itself.

QUIP: 23
Task-number: QTBUG-136184
Pick-to: 6.8
Change-Id: Ie0c5f63e3c7d4c8557dab8c12af412fd6dad1464
Reviewed-by: Oliver Eftevaag <oliver.eftevaag@qt.io>
(cherry picked from commit 2ef3de7803b9a35f58dc1355d5d6ad986fff18fc)
Reviewed-by: Qt Cherry-pick Bot <cherrypick_bot@qt-project.org>
---
 src/labs/sharedimage/qsharedimageloader.cpp      | 1 +
 src/labs/sharedimage/qsharedimageloader_p.h      | 1 +
 src/labs/sharedimage/qsharedimageprovider.cpp    | 1 +
 src/labs/sharedimage/qsharedimageprovider_p.h    | 1 +
 src/labs/sharedimage/qtlabssharedimageglobal_p.h | 1 +
 5 files changed, 5 insertions(+)

diff --git a/src/labs/sharedimage/qsharedimageloader.cpp b/src/labs/sharedimage/qsharedimageloader.cpp
index f3c17f9ebb..75648fcd21 100644
--- a/src/labs/sharedimage/qsharedimageloader.cpp
+++ b/src/labs/sharedimage/qsharedimageloader.cpp
@@ -1,5 +1,6 @@
 // Copyright (C) 2017 The Qt Company Ltd.
 // SPDX-License-Identifier: LicenseRef-Qt-Commercial OR LGPL-3.0-only OR GPL-2.0-only OR GPL-3.0-only
+// Qt-Security score:significant reason:default
 
 #include "qsharedimageloader_p.h"
 #include <private/qobject_p.h>
diff --git a/src/labs/sharedimage/qsharedimageloader_p.h b/src/labs/sharedimage/qsharedimageloader_p.h
index ca59d95585..5fb531e3fb 100644
--- a/src/labs/sharedimage/qsharedimageloader_p.h
+++ b/src/labs/sharedimage/qsharedimageloader_p.h
@@ -1,5 +1,6 @@
 // Copyright (C) 2017 The Qt Company Ltd.
 // SPDX-License-Identifier: LicenseRef-Qt-Commercial OR LGPL-3.0-only OR GPL-2.0-only OR GPL-3.0-only
+// Qt-Security score:significant reason:default
 
 #ifndef QSHAREDIMAGELOADER_H
 #define QSHAREDIMAGELOADER_H
diff --git a/src/labs/sharedimage/qsharedimageprovider.cpp b/src/labs/sharedimage/qsharedimageprovider.cpp
index 33f72447e3..a65cfb0877 100644
--- a/src/labs/sharedimage/qsharedimageprovider.cpp
+++ b/src/labs/sharedimage/qsharedimageprovider.cpp
@@ -1,5 +1,6 @@
 // Copyright (C) 2017 The Qt Company Ltd.
 // SPDX-License-Identifier: LicenseRef-Qt-Commercial OR LGPL-3.0-only OR GPL-2.0-only OR GPL-3.0-only
+// Qt-Security score:significant reason:default
 
 #include <qsharedimageprovider_p.h>
 #include <private/qquickpixmap_p.h>
diff --git a/src/labs/sharedimage/qsharedimageprovider_p.h b/src/labs/sharedimage/qsharedimageprovider_p.h
index 1888f726fd..850cc60d67 100644
--- a/src/labs/sharedimage/qsharedimageprovider_p.h
+++ b/src/labs/sharedimage/qsharedimageprovider_p.h
@@ -1,5 +1,6 @@
 // Copyright (C) 2017 The Qt Company Ltd.
 // SPDX-License-Identifier: LicenseRef-Qt-Commercial OR LGPL-3.0-only OR GPL-2.0-only OR GPL-3.0-only
+// Qt-Security score:significant reason:default
 
 #ifndef QSHAREDIMAGEPROVIDER_H
 #define QSHAREDIMAGEPROVIDER_H
diff --git a/src/labs/sharedimage/qtlabssharedimageglobal_p.h b/src/labs/sharedimage/qtlabssharedimageglobal_p.h
index 1045e9f729..ffd5479c3a 100644
--- a/src/labs/sharedimage/qtlabssharedimageglobal_p.h
+++ b/src/labs/sharedimage/qtlabssharedimageglobal_p.h
@@ -1,5 +1,6 @@
 // Copyright (C) 2021 The Qt Company Ltd.
 // SPDX-License-Identifier: LicenseRef-Qt-Commercial OR LGPL-3.0-only OR GPL-2.0-only OR GPL-3.0-only
+// Qt-Security score:insignificant
 
 #ifndef QTLABSSHAREDIMAGEGLOBAL_P_H
 #define QTLABSSHAREDIMAGEGLOBAL_P_H
-- 
2.51.2

